File: src/main/java/szewek/mcflux/special/CommandSpecialGive.java
package szewek.mcflux.special;

import net.minecraft.command.CommandBase;
import net.minecraft.command.CommandException;
import net.minecraft.command.ICommandSender;
import net.minecraft.command.WrongUsageException;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagInt;
import net.minecraft.server.MinecraftServer;
import net.minecraft.util.math.BlockPos;
import szewek.fl.FL;
import szewek.mcflux.MCFluxResources;

import javax.annotation.Nullable;
import java.util.Collections;
import java.util.List;

public class CommandSpecialGive extends CommandBase {
	@Override public String getName() {
		return "sgive";
	}

	@Override public String getUsage(ICommandSender sender) {
		return "mcflux.cmd.sgive.usage";
	}

	@Override
	public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException {
		if (args.length < 2)
			throw new WrongUsageException("mcflux.cmd.sgive.usage");
		else {
			EntityPlayer player = getEntity(server, sender, args[0], EntityPlayer.class);
			int l = parseInt(args[1]);
			SpecialEvent se = SpecialEventHandler.getEvent(l);
			if (se == null)
				throw new CommandException("mcflux.cmd.sgive.noEvent", l);
			ItemStack is = new ItemStack(MCFluxResources.SPECIAL);
			is.setTagInfo("seid", new NBTTagInt(l));
			FL.giveItemToPlayer(is, player);
		}
	}

	@Override
	public List<String> getTabCompletions(MinecraftServer server, ICommandSender sender, String[] args, @Nullable BlockPos pos) {
		return args.length == 1 ? getListOfStringsMatchingLastWord(args, server.getOnlinePlayerNames()) : Collections.emptyList();
	}
}

File: src/main/java/szewek/mcflux/special/CommandSpecialGive.kt
package szewek.mcflux.special

import net.minecraft.command.CommandBase
import net.minecraft.command.CommandException
import net.minecraft.command.ICommandSender
import net.minecraft.command.WrongUsageException
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemStack
import net.minecraft.nbt.NBTTagInt
import net.minecraft.server.MinecraftServer
import net.minecraft.util.math.BlockPos
import net.minecraftforge.items.ItemHandlerHelper
import szewek.mcflux.MCFlux

class CommandSpecialGive : CommandBase() {
	override fun getName() = "sgive"

	override fun getUsage(sender: ICommandSender) = "mcflux.cmd.sgive.usage"

	@Throws(CommandException::class)
	override fun execute(server: MinecraftServer, sender: ICommandSender, args: Array<String>) {
		if (args.size < 2)
			throw WrongUsageException("mcflux.cmd.sgive.usage")
		else {
			val player = CommandBase.getEntity(server, sender, args[0], EntityPlayer::class.java)
			val l = CommandBase.parseInt(args[1])
			SpecialEventHandler.getEvent(l) ?: throw CommandException("mcflux.cmd.sgive.noEvent", l)
			val stk = ItemStack(MCFlux.Resources.SPECIAL)
			stk.setTagInfo("seid", NBTTagInt(l))
			ItemHandlerHelper.giveItemToPlayer(player, stk, -1)
		}
	}

	override fun getTabCompletions(server: MinecraftServer, sender: ICommandSender, args: Array<String>, pos: BlockPos?): List<String> {
		return if (args.size == 1) CommandBase.getListOfStringsMatchingLastWord(args, *server.onlinePlayerNames) else emptyList()
	}
}
